aiohomekit/controller/coap/pairing.py

```diff
@@ -48,6 +50,12 @@ def __init__(
         self.connection_lock = asyncio.Condition()
         self.pairing_data = pairing_data
 
+    def _async_description_update(self, description: Optional[HomeKitService]) -> None:
+        super()._async_description_update(description)
+        async_create_task(
+            self.connection.reconnect_soon(f"{description.address}:{description.port}")
```

Comment:
For CoAP at least the IPv6 address needs to be wrapped in square brackets, e.g. "[fe80::1234:2345:3456:4567]:5683". Also, the CoAP `reconnect_soon` needs fixing to accept a `str` vs a `dict`.

Reply:
Ah cheers, i will definitely get a test env set up for this stuff soon.

Ideally we would make the IP/Port comparison here directly, without creating a task, and then only have to create a task if the IP had actually changed.

Reply:
https://github.com/Jc2k/aiohomekit/pull/151/commits/133124b684f882595a6b6722b751af9241aefefc. Can probably do better still but don't want to touch CoAP too much before i've got thing working here :)